### PR TITLE
Flow and Temperature Data added in environmental pull script

### PR DESCRIPTION
**Pull request by B-yunes**
- Flow and temperature data for each JPE tributary was added to pull environmental data script
- Flow data was pulled from CDEC and USGS (except Battle Creek, Upper Clear Creek, Lower Clear Creek, that data was provided by Ryan)
- Temp data was pulled from CDEC and USGS, also interpolated temp data was added from water_temp_regression (only for Feather and Yuba) and was bund to gage data 
- Interpolated temp data is intended to be used as historical data, and  gages will cover most up to date data (>2024) 
- "try", "if", "else" statements were added to all gage data queries to automate the alternative scenario where existing data is used (from SRJPEdata::environmental_data), instead of gage data, only when gage data queries do not work 
- Refer to Issue #12

### CODE REVIEW

File: data-raw/pull_environmental_data.R [hunk 1]
```diff
@@ -0,0 +1,683 @@
...
+  glimpse()
+
+ggplot(all_flow |> 
+         filter(statistic == "mean"),
+         aes(x= date, y = value, color=stream)) +
+  geom_line() +
+  facet_wrap(~stream)
+
+#Combine all temperature data from different streams
+all_temps <- bind_rows(battle_creek_daily_temp,
```
Comment (ErinCain) [inline data-raw/pull_environmental_data.R:666]: Stream name is inconsistent across interpolated and gage sources. Some are lower case, others are upper, and some are a mix. We only want one naming convention for each stream. We can make them all lower case for consistency.

**Review by ErinCain — COMMENTED** [review]
Hi Badhia, I added some comments on sections that need to be updated. A few big level items is that this script needs to produce an environmental_data object. So please combine the flow and temp datasets to produce that environmental data object. Additionally, please review all code documentation to make sure that it makes sense with actual code. 

And there are too many files associated with this update. We should only have the water_temp_regression.md the pull_environmental_data.R, and the interpolated files. Let's talk about how this happened and Ashley and I can clean up before merging.

**Review by alvizek — CHANGES_REQUESTED** [review]
Badhia - Erin and I reviewed this together. Erin's review contains specific changes that need to be implemented, please address these. Overall here are a few items I noticed in this script that apply to your work in general and can be improved:
- attention to detail (there were places where you forgot to do tasks that were requested and outlined in the script)
- documentation (please work to improve the comments and text you include in your scripts. comments are meant to provide context to the code are important for knowledge transfer and interpretation. please make sure comments are accurate, clean and consistent, and provide enough detail to orient someone to the code.)
- checking work (we noticed a few errors in the code which could have been easily caught by checking your work. please make sure to check your work to ensure the code is doing what you want it to do)

File: data-raw/pull_environmental_data.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R:632]: added save function, also added "overwrite = TRUE" statement

File: data-raw/pull_environmental_data.R [hunk 2]
```diff
@@ -0,0 +1,683 @@
...
+### Flow Data Pull Tests
+
+try(butte_creek_data_query <- CDECRetrieve::cdec_query(station = "BCK", dur_code = "H", sensor_num = "20", start_date = "1995-01-01"))
+# Filter existing data to use as a back up 
+butte_creek_existing_flow  <- SRJPEdata::environmental_data |> 
+  filter(gage_agency == "CDEC" & 
+           gage_number == "BCK" & 
+           parameter == "flow") |> 
+  select(-site)  
+# Confirm data pull did not error out, if does not exist - use existing flow,
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R:78]: changed text to match either flow or temperature, according to type of data pulled

File: data-raw/pull_environmental_data.R [hunk 3]
```diff
@@ -0,0 +1,683 @@
...
+           gage_number == "BCK" &
+           parameter == "temperature") |> 
+  select(-site)
+# Confirm data pull did not error out, if does not exist - use existing flow, 
+# if exists - reformat new data pull
+try(if(!exists("butte_creek_temp_query")) 
+  butte_creek_daily_temp <- butte_creek_existing_temp 
+  else(butte_creek_daily_temp <- butte_creek_temp_query |> 
+    mutate(date = as_date(datetime),
+           temp_degC = fahrenheit.to.celsius(parameter_value, round = 1)) |>
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R]: changed function to SRJPEdata::fahrenheit_to_celsius() function, deleted watermetrics library

File: data-raw/pull_environmental_data.R [hunk 4]
```diff
@@ -0,0 +1,683 @@
...
+           parameter == "temperature") |> 
+  select(-site)
+
+# Confirm data pull did not error out, if does not exist - use existing flow, 
+# if exists - reformat new data pull
+try(if(!exists("deer_creek_temp_query")) 
+  deer_creek_daily_temp <- deer_creek_existing_temp 
+  else(deer_creek_daily_temp <- deer_creek_temp_query |> 
+         mutate(date = as_date(datetime),
+                temp_degC = fahrenheit.to.celsius(parameter_value, round = 1)) |>
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R]: changed function to SRJPEdata::fahrenheit_to_celsius() function, deleted watermetrics library

File: data-raw/pull_environmental_data.R [hunk 5]
```diff
@@ -0,0 +1,683 @@
...
+              max = max(temp_degC, na.rm = TRUE),
+              min = min(temp_degC, na.rm = TRUE)) |> 
+    pivot_longer(mean:min, names_to = "statistic", values_to = "value") |>
+    mutate(stream = "Butte Creek",
+           gage_agency = "CDEC",
+           gage_number = "BCK",
+           parameter = "temperature")))
+
+### Temp Data Pull Tests  
+
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R]: added few additional temperature data pull tests to confirm that new data pull has more data

File: data-raw/pull_environmental_data.R [hunk 6]
```diff
@@ -0,0 +1,683 @@
...
+                   max = max(temp_degC, na.rm = TRUE),
+                   min = min(temp_degC, na.rm = TRUE)) |>
+         pivot_longer(mean:min, names_to = "statistic", values_to = "value") |>
+         mutate(stream = "Deer Creek",
+                gage_agency = "CDEC",
+                gage_number = "DCV",
+                parameter = "temperature")))
+
+### Temp Data Pull Tests 
+
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R]: added few additional temperature data pull tests to confirm that new data pull has more data

File: data-raw/pull_environmental_data.R [hunk 7]
```diff
@@ -0,0 +1,683 @@
...
+           parameter = "temperature")))
+
+### Temp Data Pull Tests  
+
+
+## Clear Creek
+### Flow Data Pull 
+#### Gage Agency (CDEC, IGO)
+### Flow Data Pull Tests 
+try(clear_creek_data_query <- CDECRetrieve::cdec_query(station = "IGO", dur_code = "H", sensor_num = "20", start_date = "2003-01-01"))
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R]: changes clear creek flow data for USGS 11372000 station

File: data-raw/pull_environmental_data.R [hunk 8]
```diff
@@ -0,0 +1,683 @@
...
+  else(lower_feather_river_daily_flows <- lower_feather_river_data_query |> 
+         mutate(parameter_value = ifelse(parameter_value < 0, NA_real_, parameter_value)) |> 
+         group_by(date = as.Date(datetime)) |> 
+         summarise(mean = mean(parameter_value, na.rm = TRUE),
+                   max = ifelse(all(is.na(parameter_value)), NA, max(parameter_value, na.rm = TRUE)),
+                   min = ifelse(all(is.na(parameter_value)), NA, min(parameter_value, na.rm = TRUE))) |> 
+         pivot_longer(mean:min, names_to = "statistic", values_to = "value") |> 
+         mutate(stream = "Feather River",  
+                gage_agency = "CDEC",
+                gage_number = "FBL",
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R]: changed gage number to FSB

File: data-raw/pull_environmental_data.R [hunk 9]
```diff
@@ -0,0 +1,683 @@
...
+# Do a few additional flow data pull tests to confirm that new data pull has 
+# more data
+try(if(nrow(lower_feather_river_daily_flows) < nrow(lower_feather_river_existing_flow)) 
+  lower_feather_river_daily_flows <- lower_feather_river_existing_flow)
+
+### Temp Data Pull 
+#GRL will represent the High Flow Channel (HFC) and FRA will represent the Low Flow Channel (LFC).
+
+#Interpolation feather hfc
+feather_hfc_interpolated <- read.csv(here::here("data-raw", "temperature-data", "feather_hfc_temp_interpolation.csv")) |>
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R:366]: Changed R Markdown "Evaluate chunks in directory:" setting to "project"

File: data-raw/pull_environmental_data.R [hunk 10]
```diff
@@ -0,0 +1,683 @@
...
+    pivot_longer(mean:min, names_to = "statistic", values_to = "value") |>
+    mutate(stream = "Feather River",
+           gage_agency = "CDEC",
+           gage_number = "FRA",
+           parameter = "temperature") |> 
+    bind_rows(feather_lfc_interpolated)
+   ))
+
+### Temp Data Pull Tests 
+
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R]: added few additional temperature data pull tests to confirm that new data pull has more data

File: data-raw/pull_environmental_data.R [hunk 11]
```diff
@@ -0,0 +1,683 @@
...
+                gage_number = "11381500",
+                parameter = "flow",
+                statistic = "mean" 
+         )))
+
+# Do a few additional flow data pull tests to confirm that new data pull has 
+# more data
+try(if(nrow(mill_creek_daily_flows) < nrow(mill_creek_existing_flow)) 
+  mill_creek_daily_flows <- mill_creek_existing_flow)
+
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R:451]: changed spacing throughout the code to keep it consistent

File: data-raw/pull_environmental_data.R [hunk 12]
```diff
@@ -0,0 +1,683 @@
...
+                   min = min(temp_degC, na.rm = TRUE)) |> 
+         pivot_longer(mean:min, names_to = "statistic", values_to = "value") |>
+         mutate(stream = "Mill creek",
+                gage_agency = "CDEC",
+                gage_number = "MLM",
+                parameter = "temperature")))
+
+
+### Temp Data Pull Tests 
+
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R]: added few additional temperature data pull tests to confirm that new data pull has more data

File: data-raw/pull_environmental_data.R [hunk 13]
```diff
@@ -0,0 +1,683 @@
...
+  select(-site)
+
+
+# Confirm data pull did not error out, if does not exist - use existing flow, 
+# if exists - reformat new data pull
+try(if(!exists("mill_creek_temp_query")) 
+  mill_creek_daily_temp <- mill_creek_existing_temp 
+  else(mill_creek_daily_temp <- mill_creek_temp_query |> 
+         mutate(date = as_date(datetime),
+                temp_degC = fahrenheit.to.celsius(parameter_value, round = 1)) |>
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R]: changed function to SRJPEdata::fahrenheit_to_celsius() function, deleted watermetrics library

File: data-raw/pull_environmental_data.R [hunk 14]
```diff
@@ -0,0 +1,683 @@
...
+  select(-site)
+# Confirm data pull did not error out, if does not exist - use existing flow, 
+# if exists - reformat new data pull
+try(if(!exists("sac_river_temp_query")) 
+  sac_river_daily_temp <- sac_river_existing_temp 
+  else(sac_river_daily_temp <- sac_river_temp_query |> 
+         select(Date, temp_degC =  X_00010_00003) %>%
+         as_tibble() %>% 
+         rename(date = Date,
+                parameter = temp_degC) %>%
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R]: added missing value = temp_degC

File: data-raw/pull_environmental_data.R [hunk 15]
```diff
@@ -0,0 +1,683 @@
...
+                parameter = temp_degC) %>% 
+         mutate(stream = "Sacramento River",
+                       gage_agency = "USGS",
+                       gage_number = "11390500",
+                       parameter = "temperature")))
+
+
+
+### Temp Data Pull Tests 
+
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R]: added few additional temperature data pull tests to confirm that new data pull has more data

File: data-raw/pull_environmental_data.R [hunk 16]
```diff
@@ -0,0 +1,683 @@
...
+                parameter = "flow",
+                statistic = "mean"
+         )))
+
+# Do a few additional flow data pull tests to confirm that new data pull has 
+# more data
+try(if(nrow(yuba_daily_flows) < nrow(yuba_existing_flow)) 
+  yuba_daily_flows <- yuba_existing_flow)
+
+yuba_river_existing_temp <- SRJPEdata::environmental_data |>
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R]: deleted duplicate temp pull code for Yuba river temperature

File: data-raw/pull_environmental_data.R [hunk 17]
```diff
@@ -0,0 +1,683 @@
...
+
+yuba_river_existing_temp <- SRJPEdata::environmental_data |> 
+  filter(gage_agency == "CDEC" &
+           gage_number == "YR7" &
+           parameter == "temperature") |> 
+  select(-site)
+
+
+#pull temp data from CDEC YR7
+try(yuba_river_temp_query <- cdec_query(station = "YR7", dur_code = "E", sensor_num = "146", start_date = "2019-01-01"))
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R]: change Yuba River Temp query to start data of "2024-02-07", which is the last date of the interpolated temp data

File: data-raw/pull_environmental_data.R [hunk 18]
```diff
@@ -0,0 +1,683 @@
...
+
+#Combine all flow data from different streams
+all_flow <- bind_rows(battle_creek_daily_flows,
+                      butte_creek_daily_flows, 
+                      clear_creek_daily_flows,
+                      deer_creek_daily_flows,
+                      feather_hfc_river_daily_flows,
+                      feather_lfc_river_daily_flows,
+                      lower_feather_river_daily_flows,
+                      feather_lfc_river_daily_flows,
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply data-raw/pull_environmental_data.R]: removed duplicated feather_river_lfc_daily_flows in bind_flows list

File: data-raw/pull_environmental_data.R (hunk 1 above)
Comment (B-yunes) [reply data-raw/pull_environmental_data.R:638]: changed stream names to be consistent throughout the script to all lower case

File: _pkgdown.yml [hunk 19]
```diff
@@ -16,6 +16,7 @@ reference:
   desc: Data used as model inputs to passage to spawner (p2s) model or directly input to stock recruit
```
[the remark this answers is not in the document]
Comment (B-yunes) [reply _pkgdown.yml:16]: Talked through this with Erin, noted some better practices to apply from now on to prevent this from happening again

**Comment by ErinCain** [comment]
moved logic over to another branch to clean up excess files.

**Pull request closed without merging**